api/environments/dynamodb/services.py

```diff
@@ -26,39 +45,55 @@ def migrate_environments_to_v2(project_id: int) -> IdentityOverridesV2Changeset
 
     logger.info("Migrating environments to v2 for project %d", project_id)
 
-    environments_to_migrate = Environment.objects.filter(project_id=project_id)
-    identity_overrides_to_migrate = _iter_paginated_overrides(
-        identity_wrapper=identity_wrapper,
-        environments=environments_to_migrate,
+    environments_to_migrate = Environment.objects.filter_for_document_builder(
+        project_id=project_id
     )
 
-    changeset = IdentityOverridesV2Changeset(
-        to_put=list(identity_overrides_to_migrate), to_delete=[]
-    )
-    logger.info(
-        "Retrieved %d identity overrides to migrate for project %d",
-        len(changeset.to_put),
-        project_id,
-    )
+    identity_overrides_changeset = IdentityOverridesV2Changeset(to_put=[], to_delete=[])
+    result_status = EdgeV2MigrationStatus.COMPLETE
+
+    try:
+        to_put = list(
```

Comment:
Do we have a rough idea about how much data will this be for an environment with several million identities? I just want to make sure we don't hurt task processor

Reply:
`len(to_put)` is equal to the count of only the identities that have overridden feature states. The rest of identities is iterated over but discarded.

Reply:
But are we still fetching all the identities? I thought we did a find a way only fetch identities that have overrides?

Reply:
No, we haven't. Even if we use a filter expression we'll spend the same amount of read capacity.

Reply:
Let me compare performance between the two variants. If there's no major difference, I'm in favour of this change.

Reply:
Done.